src/relays/TrustedRelay.sol

```diff
@@ -137,9 +140,25 @@ contract TrustedRelay {
         emit SignersRemoved(signers_);
     }
 
+    function addRelayers(address[] calldata relayers_) external auth {
+        for(uint256 i; i < relayers_.length; i++) {
+            relayers[relayers_[i]] = 1;
+        }
+        emit RelayersAdded(relayers_);
+    }
+
+    function removeRelayers(address[] calldata relayers_) external auth {
+        for(uint256 i; i < relayers_.length; i++) {
+            relayers[relayers_[i]] = 0;
+        }
+        emit RelayersRemoved(relayers_);
+    }
```

Comment:
Wondering if we want to add a `isTrustedForwarder` function as in the eip (https://eips.ethereum.org/EIPS/eip-2771). See for example in - https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/metatx/ERC2771Context.sol.

I don't yet understand the full flow here (relayer vs signer, mev relevance, is basicRelay now also "trusted"?), so don't have a strong opinion.

Reply:
Gelato only loosely follows EIP2771. For example it doesn't use a TrustedForwarder contract to verify the signature (hence we verify the signature ourselves in BasicRelay). So I think it's fine to not strictly adhere to the EIP here. We only follow the EIP in the way the feeCollector is recovered. Will clarify in comments.

I'd say that BasicRelay is not trusted in so far as the user is in control of all the signed parameters (gasFee, expiry, maxFeePercentage) unlike in TrustedRelay where these parameters are signed by some trusted backend key. That being said, it is no longer permissionless as BasicRelay can now only be called by whitelisted relayer accounts (to mitigate MEV).